File: src/main/java/com/examples/counterobject/Counter.java
package com.examples.counterobject;

import edu.illinois.cs.cs125.questioner.lib.Correct;

/*
 * Define and implement a `public` class `Counter`.
 * `Counter` should provide a single public constructor that accepts an `int` parameter.
 * It should also provide `increment` and `decrement` methods, both taking no parameters and returning `void`.
 * `increment` should increase the stored count, while `decrement` should decrease it.
 * Finally, provide a `getValue` method that returns the stored value.
 */

@Correct(name = "Counter", author = "challen@illinois.edu", version = "2021.6.0")
public class Counter {
  private int value;

  public Counter(int setValue) {
    value = setValue;
  }

  public void increment() {
    value++;
  }

  public void decrement() {
    value--;
  }

  public int getValue() {
    return value;
  }
}

File: src/main/java/com/examples/counterobject/correct/kotlin/Counter.kt
package com.examples.counterobject.correct.kotlin

import edu.illinois.cs.cs125.questioner.lib.AlsoCorrect

/*
 * Define and implement a class `Counter`.
 * `Counter` should provide primary constructor that accepts an `Int` parameter.
 * It should also provide `increment` and `decrement` methods, both taking no parameters and returning `Unit`.
 * `increment` should increase the stored count, while `decrement` should decrease it.
 * Finally, provide a `getValue` method that returns the stored value.
 */

@AlsoCorrect
class Counter(setValue: Int) {
  var value = setValue
    private set
  fun increment() {
    value++
  }
  fun decrement() {
    value--
  }
}